gcn_package/models/gcn.py

```diff
@@ -34,8 +36,8 @@ def forward(self, x):
         x = F.relu(x)
         x = self.conv6(x, self.edge_index, self.edge_weight)
         x = tg.nn.global_mean_pool(x,torch.from_numpy(np.array(range(x.size(0)),dtype=int)))
-        
-        x = x.view(-1, 512*32)
+
+        x = x.view(-1, self.batch_size * self.n_roi)
```

Comment:
batch_size is confusing here. That does not make sense that the model depends on data generator batch size, it should be independent.
The best way to flatten the actual output is to simply use:
x = torch.flatten(x, 1) (please double check dimension here)
This way batch_size is not needed

Reply:
I can modify this line to what you suggested here